Error on 1.19.3

Hello, I was looking again over it and I got this error from the server log:
I am not able to see any loot tables.

**First:**
[REI] ReltServer of ReltServer [roughly_enough_loot_tables] plugin failed to reloadable-plugin/EntryTypeRegistryImpl/!
java.lang.ExceptionInInitializerError: null
	at Not Enough Crashes deobfuscated stack trace.(1.19.3+build.4) ~[?:?]
	at griglog.relt.rei_plugin.ReltServer.registerEntryTypes(ReltPlugin.kt:37) ~[RoughlyEnoughLootTables-1.19-1.0.jar:?]
	at me.shedaniel.rei.impl.common.entry.type.EntryTypeRegistryImpl.acceptPlugin(EntryTypeRegistryImpl.java:56) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.api.common.registry.Reloadable.acceptPlugin(Reloadable.java:72) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$startReload$9(PluginManagerImpl.java:340) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$pluginSection$4(PluginManagerImpl.java:198) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$startReload$10(PluginManagerImpl.java:330) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.pluginSection(PluginManagerImpl.java:196) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.startReload(PluginManagerImpl.java:328) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore._reloadPlugins(RoughlyEnoughItemsCore.java:123) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore._reloadPlugins(RoughlyEnoughItemsCore.java:114) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore.lambda$onInitialize$3(RoughlyEnoughItemsCore.java:144) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.resource.SimpleResourceReload.method_18365(SimpleResourceReload:69) ~[?:?]
	at net.minecraft.util.Util.waitAndApply(Util:948) ~[?:?]
	at net.minecraft.util.Util.waitAndApply(Util:936) ~[?:?]
	at net.minecraft.server.Main.main(Main:150) ~[server-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[minecraft_server.jar:0.11.1]
Caused by: java.lang.RuntimeException: Cannot load class me.shedaniel.rei.api.client.entry.renderer.EntryRenderer in environment type SERVER
	at net.fabricmc.loader.impl.transformer.FabricTransformer.transform(FabricTransformer.java:59) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPreMixinClassByteArray(KnotClassDelegate.java:462) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:415) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1012) ~[?:?]
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.defineClassFwd(KnotClassLoader.java:153) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:355) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at griglog.relt.rei_plugin.TableEntryDef.<clinit>(TableEntryDef.kt:28) ~[RoughlyEnoughLootTables-1.19-1.0.jar:?]
	... 22 more

**A bit later:**
[REI] ReltServer of ReltServer [roughly_enough_loot_tables] plugin failed to reloadable-plugin/EntryTypeRegistryImpl/!
java.lang.NoClassDefFoundError: Could not initialize class griglog.relt.rei_plugin.TableEntryDef
	at Not Enough Crashes deobfuscated stack trace.(1.19.3+build.4) ~[?:?]
	at griglog.relt.rei_plugin.ReltServer.registerEntryTypes(ReltPlugin.kt:37) ~[RoughlyEnoughLootTables-1.19-1.0.jar:?]
	at me.shedaniel.rei.impl.common.entry.type.EntryTypeRegistryImpl.acceptPlugin(EntryTypeRegistryImpl.java:56) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.api.common.registry.Reloadable.acceptPlugin(Reloadable.java:72) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$startReload$9(PluginManagerImpl.java:340) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$pluginSection$4(PluginManagerImpl.java:198) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.lambda$startReload$10(PluginManagerImpl.java:330) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.pluginSection(PluginManagerImpl.java:196) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.impl.common.plugins.PluginManagerImpl.startReload(PluginManagerImpl.java:328) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore._reloadPlugins(RoughlyEnoughItemsCore.java:123) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore._reloadPlugins(RoughlyEnoughItemsCore.java:114) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at me.shedaniel.rei.RoughlyEnoughItemsCore.lambda$onInitialize$3(RoughlyEnoughItemsCore.java:144) ~[RoughlyEnoughItems-10.0.581.jar:?]
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.resource.SimpleResourceReload.method_18365(SimpleResourceReload:69) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,6 +1,7 @@
 val modVersion: String by project
 val mavenGroup: String by project
 val minecraftVersion: String by project
+val parchmentVersion: String by project
 val yarnMappings: String by project
 val loaderVersion: String by project
 val fabricVersion: String by project
@@ -41,7 +42,7 @@ dependencies {
     //mappings("net.fabricmc", "yarn", yarnMappings, null, "v2")
     mappings(loom.layered() {
         officialMojangMappings()
-        parchment("org.parchmentmc.data:parchment-1.18.2:2022.08.21@zip")
+        parchment("org.parchmentmc.data:parchment-${minecraftVersion}:${parchmentVersion}@zip")
     })
 
     modImplementation("net.fabricmc:fabric-loader:${loaderVersion}")

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,21 +3,21 @@ org.gradle.jvmargs = -Xmx1G
 org.gradle.warning.mode = all
 
 # Check these on https://fabricmc.net/develop/
-minecraftVersion=1.19.2
-loaderVersion=0.14.9
+minecraftVersion=1.19.3
+loaderVersion=0.14.17
 #Fabric api
-fabricVersion=0.60.0+1.19.2
-loomVersion = 1.0-SNAPSHOT
+fabricVersion=0.76.0+1.19.3
+loomVersion = 1.1-SNAPSHOT
 
 # Mod Properties
-modVersion = 1.19-1.0
+modVersion = 1.19.3-1.0
 mavenGroup = griglog
 archivesBaseName = RoughlyEnoughLootTables
 
 # Kotlin
-systemProp.kotlinVersion = 1.7.10
-fabricKotlinVersion = 1.8.3+kotlin.1.7.10
+systemProp.kotlinVersion = 1.8.10
+fabricKotlinVersion = 1.9.2+kotlin.1.8.10
 
 #Dependecies
-reiVersion = 9.1.528
-parchmentVersion = 2022.08.28
+reiVersion = 10.0.596
+parchmentVersion = 2023.03.12

--- a/src/main/kotlin/griglog/relt/entry_points/ReltClient.kt
+++ b/src/main/kotlin/griglog/relt/entry_points/ReltClient.kt
@@ -3,10 +3,7 @@ package griglog.relt.entry_points
 import griglog.relt.RELT
 import griglog.relt.table_storage.recieveLootTables
 import net.fabricmc.api.ClientModInitializer
-import net.fabricmc.api.ModInitializer
 import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking
-import java.awt.Desktop
-import java.awt.GraphicsEnvironment
 
 class ReltClient : ClientModInitializer {
     override fun onInitializeClient() {

--- a/src/main/kotlin/griglog/relt/rei_plugin/ReltPlugin.kt
+++ b/src/main/kotlin/griglog/relt/rei_plugin/ReltPlugin.kt
@@ -9,7 +9,7 @@ import me.shedaniel.rei.api.common.entry.EntryIngredient
 import me.shedaniel.rei.api.common.entry.type.EntryTypeRegistry
 import me.shedaniel.rei.api.common.plugins.REIServerPlugin
 import me.shedaniel.rei.api.common.util.EntryIngredients
-import net.minecraft.core.Registry
+import net.minecraft.core.registries.BuiltInRegistries
 import net.minecraft.resources.ResourceLocation
 import net.minecraft.tags.ItemTags
 import net.minecraft.world.item.FishingRodItem
@@ -42,7 +42,7 @@ fun getInputs(name: ResourceLocation, table: LootTable): EntryIngredient?{
     if (table.paramSet == LootContextParamSets.PIGLIN_BARTER)
         return EntryIngredients.ofItemTag(ItemTags.PIGLIN_LOVED)
     if (table.paramSet == LootContextParamSets.FISHING)
-        return EntryIngredients.ofItemStacks(Registry.ITEM.filter{it is FishingRodItem}.map{ItemStack(it)})
+        return EntryIngredients.ofItemStacks(BuiltInRegistries.ITEM.filter{it is FishingRodItem}.map{ItemStack(it)})
     if (table.paramSet == LootContextParamSets.CHEST)
         return EntryIngredients.of(ItemStack(Items.CHEST))
     return null

--- a/src/main/kotlin/griglog/relt/rei_plugin/TableDisplay.kt
+++ b/src/main/kotlin/griglog/relt/rei_plugin/TableDisplay.kt
@@ -5,7 +5,7 @@ import me.shedaniel.rei.api.common.display.basic.BasicDisplay
 import me.shedaniel.rei.api.common.entry.EntryIngredient
 import me.shedaniel.rei.api.common.entry.EntryStack
 import me.shedaniel.rei.api.common.util.EntryIngredients
-import net.minecraft.core.Registry
+import net.minecraft.core.registries.BuiltInRegistries
 import net.minecraft.resources.ResourceLocation
 import java.util.*
 
@@ -13,7 +13,7 @@ class TableDisplay : BasicDisplay {
     constructor(name: ResourceLocation, result: Collection<ItemLike>, tables:Collection<EntryStack<ResourceLocation>>, inputs: EntryIngredient?) : super(
         mutableListOf(tableIng(name), inputs).filterNotNull(),
         mutableListOf<EntryIngredient>().apply{
-            addAll(result.sortedWith(Comparator.comparingInt{il -> Registry.ITEM.getId(il.stack.item)})
+            addAll(result.sortedWith(Comparator.comparingInt{il -> BuiltInRegistries.ITEM.getId(il.stack.item)})
                 .map{EntryIngredients.ofItemStacks(it.genStacks())})
             addAll(tables.map{EntryIngredient.of(it)})
         },

--- a/src/main/kotlin/griglog/relt/table_resolving/ItemLike.kt
+++ b/src/main/kotlin/griglog/relt/table_resolving/ItemLike.kt
@@ -1,29 +1,28 @@
 package griglog.relt.table_resolving
 
 import griglog.relt.wrapHoverName
-import net.minecraft.core.Registry
+import net.minecraft.core.registries.BuiltInRegistries
 import net.minecraft.world.item.*
 import net.minecraft.world.item.enchantment.Enchantment
-import net.minecraft.world.item.enchantment.EnchantmentHelper
 import net.minecraft.world.item.enchantment.EnchantmentInstance
 import net.minecraft.world.level.storage.loot.providers.number.NumberProvider
 import java.util.*
 
 private fun enchSet() = TreeSet<EnchantmentInstance> { a, b ->
     if (a.enchantment != b.enchantment)
-        Registry.ENCHANTMENT.getId(a.enchantment).compareTo(Registry.ENCHANTMENT.getId(b.enchantment))
+        BuiltInRegistries.ENCHANTMENT.getId(a.enchantment).compareTo(BuiltInRegistries.ENCHANTMENT.getId(b.enchantment))
     else
         a.level.compareTo(b.level)
 }
 
 private val allDiscoverableEnchants = enchSet().apply{
-    for (ench in Registry.ENCHANTMENT.filter{it.isDiscoverable}){
+    for (ench in BuiltInRegistries.ENCHANTMENT.filter{it.isDiscoverable}){
         for (lvl in ench.minLevel..ench.maxLevel)
             add(EnchantmentInstance(ench, lvl))
     }
 }
 
-//todo: in case of poor performance, cache Registry.Item.getId(stack.item)
+//todo: in case of poor performance, cache BuiltInRegistries.Item.getId(stack.item)
 class ItemLike{
     constructor(item: Item){
         stack = ItemStack(item)
@@ -42,7 +41,7 @@ class ItemLike{
         if (minPoints <= 0) minPoints = 1
         if (maxPoints <= 0) maxPoints = 1
 
-        for (ench in Registry.ENCHANTMENT){
+        for (ench in BuiltInRegistries.ENCHANTMENT){
             if ((ench.isTreasureOnly && !treasure)
                 || !ench.isDiscoverable
                 || (stack.item != Items.BOOK && !ench.category.canEnchant(stack.item)))

--- a/src/main/kotlin/griglog/relt/table_resolving/Tables.kt
+++ b/src/main/kotlin/griglog/relt/table_resolving/Tables.kt
@@ -2,17 +2,13 @@ package griglog.relt.table_resolving
 
 import griglog.relt.rei_plugin.TableEntryDef
 import me.shedaniel.rei.api.common.entry.EntryStack
-import net.minecraft.core.Registry
+import net.minecraft.core.registries.BuiltInRegistries
 import net.minecraft.resources.ResourceLocation
 import net.minecraft.world.item.Item
 import net.minecraft.world.item.Items
 import net.minecraft.world.item.alchemy.PotionUtils
 import net.minecraft.world.level.storage.loot.LootTable
-import net.minecraft.world.level.storage.loot.entries.CompositeEntryBase
-import net.minecraft.world.level.storage.loot.entries.LootItem
-import net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer
-import net.minecraft.world.level.storage.loot.entries.LootTableReference
-import net.minecraft.world.level.storage.loot.entries.TagEntry
+import net.minecraft.world.level.storage.loot.entries.*
 import net.minecraft.world.level.storage.loot.functions.*
 
 
@@ -31,7 +27,7 @@ fun resolveEntry(entry: LootPoolEntryContainer, items: MutableSet<ItemLike>, tab
     when(entry){
         is LootItem -> resolveItem(entry.item, entry.functions)?.let{items.add(it)}
         is LootTableReference -> tables.add(EntryStack.of(TableEntryDef.type, entry.name))
-        is TagEntry -> Registry.ITEM.getTagOrEmpty(entry.tag).forEach{item ->
+        is TagEntry -> BuiltInRegistries.ITEM.getTagOrEmpty(entry.tag).forEach{ item ->
             resolveItem(item.value(), entry.functions)?.let{items.add(it)}
         }
         is CompositeEntryBase -> entry.children.forEach{resolveEntry(it, items, tables)}

--- a/src/main/kotlin/griglog/relt/table_storage/ServerTables.kt
+++ b/src/main/kotlin/griglog/relt/table_storage/ServerTables.kt
@@ -27,7 +27,7 @@ fun tryUpdateLootTables(server: MinecraftServer) {
     /*
     for (i in 0..50) {
         val test: LootTable = LootTable.lootTable().pool(LootPool.lootPool().also {
-            for (item in Registry.ITEM) {
+            for (item in BuiltInRegistries.ITEM) {
                 it.with(LootItem.lootTableItem(item).build())
             }
         }.build()).build()

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -27,12 +27,12 @@
   "accessWidener" : "roughly_enough_loot_tables.accesswidener",
   "depends": {
     "java": ">=17",
-    "minecraft": ">=1.19.2",
-    "fabricloader": ">=0.14.9",
-    "fabric": "*",
-    "fabric-language-kotlin": ">=1.8.1+kotlin.1.7.10"
+    "minecraft": ">=1.19.3",
+    "fabricloader": ">=0.14.17",
+    "fabric-api": "*",
+    "fabric-language-kotlin": ">=1.9.2+kotlin.1.8.10"
   },
   "suggests": {
-    "roughlyenoughitems": ">=9.1"
+    "roughlyenoughitems": ">=10.0"
   }
 }

